lang/java/maven-plugin/src/main/java/org/apache/avro/mojo/AbstractAvroMojo.java

```diff
@@ -213,7 +213,9 @@ public void execute() throws MojoExecutionException {
     if (hasImports) {
       for (String importedFile : imports) {
         File file = new File(importedFile);
-        if (file.isDirectory()) {
+        if (!file.exists()) {
```

Comment:
Nice !!
Same logic should apply to getIncludedFiles method, line 259 ?
(or may be put this control in another method as "checkImportFiles", as it concerns only "imports" variables.

Reply:
I was considering that, but it seems that if _imports_ field is non-null, then we may be sure that during the execution we will check every imported file in line 216. Indeed, even if _getIncludedFiles_ would execute first, the nonexistent file will be found by line 216 in a next iteration. Do you think it's anyway beneficial to check it anyway in _getIncludedFiles_?

Reply:
On line 256, getIncludedFiles method scan for all imports ...
```
      for (String importFile : this.imports) {
        File file = new File(importFile);
  ...
```
So, if you have 2 files, and only the first exists, this second method will encountered this unexisting file before execute method. So, even if used method (isFile & isDirectory) won't fails, i found that checking before using cleaner.

Reply:
Of course I understand that - I just pointed out that _getIncludedFiles_ touches this file earlier, but with no real downside. Anyway, I moved existence check to the separate method that checks everything in the first place - will that work for you?